### PR TITLE
Fixed NPE in validateHMAC

### PR DESCRIPTION
**Pull request by abhilash-adyen**
**Description**
Fixed NPE in `validateHMAC`

**Tested scenarios**

**Fixed issue**:  
#554

### CODE REVIEW

File: src/test/java/com/adyen/util/HMACValidatorTest.java [hunk 1]
```diff
@@ -27,7 +27,9 @@
 
 import java.security.SignatureException;
 
+import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
```
Comment (Aleffio) [inline src/test/java/com/adyen/util/HMACValidatorTest.java:32]:
```suggestion
```

Comment (abhilash-adyen) [reply src/test/java/com/adyen/util/HMACValidatorTest.java]: Fixed.

**Review by Aleffio — APPROVED** [review]

File: src/main/java/com/adyen/util/HMACValidator.java [hunk 2]
```diff
@@ -65,7 +65,7 @@ public String calculateHMAC(String data, String key) throws java.security.Signat
     }
 
     // To calculate the HMAC SHA-256
-    public String calculateHMAC(NotificationRequestItem notificationRequestItem, String key) throws SignatureException {
```
Comment (rkewlani) [inline src/main/java/com/adyen/util/HMACValidator.java:68]: possible NPE on line#57

File: src/main/java/com/adyen/util/HMACValidator.java [hunk 3]
```diff
@@ -79,16 +79,35 @@ public boolean validateHMAC(NotificationRequestItem notificationRequestItem, Str
         return MessageDigest.isEqual(merchantSign, expectedSign);
     }
 
-    public String getDataToSign(NotificationRequestItem notificationRequestItem) {
```
Comment (rkewlani) [inline src/main/java/com/adyen/util/HMACValidator.java:82]: calculateHMAC is checking for notificationRequestItem null but NPE will occur at #73 if notificationRequestItem is null.

File: src/main/java/com/adyen/util/HMACValidator.java (hunk 2 above)
Comment (abhilash-adyen) [reply src/main/java/com/adyen/util/HMACValidator.java:68]: The reference to `data` on line#57 has a handler for `Exception` around it, so I think it should be fine.

File: src/main/java/com/adyen/util/HMACValidator.java (hunk 3 above)
Comment (abhilash-adyen) [reply src/main/java/com/adyen/util/HMACValidator.java:82]: Fixed

**Review by Aleffio — DISMISSED** [review]

**Review by rkewlani — APPROVED** [review]

**Review by Aleffio — APPROVED** [review]

**Pull request merged**